### PR TITLE
Special ions annotations.

### PR DESCRIPTION
**Pull request by rita-gwen**

### CODE REVIEW

File: pwiz_tools/Skyline/TestFunctional/FullScanGraphTest.cs [hunk 1]
```diff
@@ -193,7 +193,9 @@ protected override void DoTest()
             Assert.AreEqual(testRange, SkylineWindow.GraphFullScan.Range);
 
             RunUI(() => SkylineWindow.SynchMzScale(SkylineWindow.GraphSpectrum, false)); // Sync from the library match to the full scan viewer
-            TestSpecialIonsAnnotations();
+            //annotations are not shown in the offscreen mode
+            if (!Skyline.Program.SkylineOffscreen)
```
Comment (brendanx67) [inline pwiz_tools/Skyline/TestFunctional/FullScanGraphTest.cs:197]: This seems overly drastic. If it is a painting issue, then you should add a method that tells what would be annotated during a paint, and use that to make sure the graph understands what it is supposed to do. We use this approach a lot with our graphs.

File: pwiz_tools/Skyline/Model/Lib/SpectrumRanker.cs [hunk 2]
```diff
@@ -701,6 +719,10 @@ private RankedMI CalculateRank(RankingState rankingState, RankedMI rankedMI)
                     continue;
                 }
 
+                //custom ions have been already matched above. No need to do anything.
```
Comment (brendanx67) [inline pwiz_tools/Skyline/Model/Lib/SpectrumRanker.cs:722]: Your comment style is at odds with the project. Put a space after the // and capitalize the first letter.

File: pwiz_tools/Skyline/Skyline.Designer.cs [hunk 3]
```diff
@@ -1,2821 +1,2831 @@
-
```
Comment (brendanx67) [inline pwiz_tools/Skyline/Skyline.Designer.cs:1]: Avoid whole file diffs, likely the result of line endings changing.

File: pwiz_tools/Skyline/SkylineGraphs.cs [hunk 4]
```diff
@@ -914,7 +918,7 @@ public void ShowLosses(IEnumerable<string> losses)
         private void IonTypeSelector_IonTypeChanges(IonType type, bool show)
         {
             switch (type)
-            {
+        {
```
Comment (brendanx67) [inline pwiz_tools/Skyline/SkylineGraphs.cs:921]: This brace indenting change seems unnecessary and incorrect.

File: pwiz_tools/Skyline/SkylineGraphs.cs [hunk 5]
```diff
@@ -957,6 +961,11 @@ private void precursorIonMenuItem_Click(object sender, EventArgs e)
...
         }
 
+        private void specialionsContextMenuItem_Click(object sender, EventArgs e)
+        {
+            ShowSpecialIons(!_graphSpectrumSettings.ShowSpecialIons);
+        }
+        
         public void SynchMzScaleToolStripMenuItemClick(IMzScalePlot source = null)
         {
             if(ListMzScaleCopyables().Count() < 2)
```
Comment (brendanx67) [inline pwiz_tools/Skyline/SkylineGraphs.cs:971]: If statements get spaces between if and (. They are not function calls.

File: pwiz_tools/Skyline/SkylineGraphs.cs [hunk 6]
```diff
@@ -984,9 +993,9 @@ private void synchMzScaleToolStripMenuItem_Click(object sender, EventArgs e)
         }
 
         //Testing support
```
Comment (brendanx67) [inline pwiz_tools/Skyline/SkylineGraphs.cs:995]: Again put a space between // and the next character.

File: pwiz_tools/Skyline/SkylineGraphs.cs [hunk 7]
```diff
@@ -995,7 +1004,7 @@ public void chargesMenuItem_DropDownOpening(object sender, EventArgs e)
             if (chargesContextMenuItem.DropDownItems.Count > 0 && chargesContextMenuItem.DropDownItems[0] is MenuControl<ChargeSelectionPanel> chargeSelector)
             {
                 chargeSelector.Update(_graphSpectrumSettings, DocumentUI.Settings.PeptideSettings);
-            }
+        }
```
Comment (brendanx67) [inline pwiz_tools/Skyline/SkylineGraphs.cs:1007]: This indenting change seems incorrect.

File: pwiz_tools/Skyline/Controls/Graphs/SpectrumGraphItem.cs [hunk 8]
```diff
@@ -1,513 +1,520 @@
-﻿/*
```
[the remark this answers is not in the document]
Comment (rita-gwen) [reply pwiz_tools/Skyline/Controls/Graphs/SpectrumGraphItem.cs:1]: I'm not sure it's a real issue.  I cannot identify a commit that introduced this whole file change. When I look at the changes in each individual commit in this PR they all show only individual line-by-line changes. When I checkout the original file and compare with it I do not see any global differences either. Both files have CR/LF line endings. How do we debug such things?

**Pull request merged**